girder/api/v1/item.py

```diff
@@ -172,7 +175,32 @@ def setMetadata(self, item, metadata, params):
                 raise RestException(
                     'Invalid key %s: keys must not start with the "$" character.' % k)
 
-        return self.model('item').setMetadata(item, metadata)
+        return self.model('item').setMetadata(item, metadata, params['allowNull'])
```

Comment:
Let's add `allowNull` as an argument to `setMetadata` so we don't need to look it up in the params dict.